dlt/common/storages/configuration.py

```diff
@@ -113,11 +113,12 @@ def protocol(self) -> str:
 
     def on_resolved(self) -> None:
         url = urlparse(self.bucket_url)
-        if not url.path and not url.netloc:
-            raise ConfigurationValueError(
-                "File path or netloc missing. Field bucket_url of FilesystemClientConfiguration"
-                " must contain valid url with a path or host:password component."
-            )
+        if not url.scheme in ("gitpythonfs", "github", "git", "s3"):
```

Comment:
> 1. pls use `not in` operator. linter will complain somewhere
Accepted

> 2. why s3 is exempt? AFAIK it must contain bucket name which is a path component

You are right. I will take `s3` out of the the exception list.

My thinking was (would need a separate github issue):
s3fs already gives nice error messages for invalid bucket name, unauthorised for bucket etc. My guess is that adlfs an , gcsfs do too.  I doubt the rule saves users from confusion/harm.  I was heading towards proving that, and then **removing this rule entirely** or only testing for specific known problem protocols (not exclusion list).  I was trying to reduce fsspec implementation-specific conditional logic in the dlt codebase.   

> also could you remind me when `gitpythonfs` does not need netloc and path

`gitpythonfs://`  returns the root of the repo.   Expected use case:  `bucket_url = "gitpythonfs://", glob = "**.md"` .    

We decided it's neater and more common to put repo path in the config (kwargs), rather than stuff it into the url netloc.    I looked at `dropboxdrivefs` and I think it will be same pattern as this - dropbox knows which account based on auth token, then the rest is globbable paths.  

At some point we could extend public  filesystem source/resource API with methods that just have `url` parameter (which can include glob - `s3://bucket/docs/**.md) instead of separate bucket_url and glob parameters.   The former is much more fsspec-onic :)